Source/WebCore/Modules/webcodecs/WebCodecsAudioDecoder.cpp

```diff
@@ -234,6 +233,7 @@ void WebCodecsAudioDecoder::isConfigSupported(ScriptExecutionContext& context, W
 
 ExceptionOr<void> WebCodecsAudioDecoder::closeDecoder(Exception&& exception)
 {
+    Ref pendingActivityKeepingJSWrapperAlive = makePendingActivity(*this);
```

Comment:
I am not sure why this is needed here.
Also the error callback is called last here, it seems the caller might be interested in taking a pending activity maybe, which probably is often the case.

Reply:
We can move to the caller if you prefer, I just added it here since there are quite a few places where we call `closeDecoder` (and same for the encoders) and this cuts down on the number of calls to `makePendingActivity`.

Reply:
I actually removed these pending activities entirely. It turns out I was just being overly cautious in the face of state changes. It's not necessary.